EIPS/eip-6888.md

```diff
@@ -0,0 +1,88 @@
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Starting from `BLOCK >= HARDFORK_BLOCK`
+
+Two new flags are added to the EVM state: unsigned error (`uerr`) and signed overferror (`serr`).
+
+From the point foward `a`, `b` and `c` references the two arguments in a math opeartion and `res` the output.
+
+It is define two new flags `uerr` and `serr` and both start as zero.
+
+The `uerr` flag "MUST" be set in the following circumstances:
+
+ - When opcode is `ADD` (`0x01`) and `res < a`
+ - When opcode is `SUB` (`0x03`) and `b > a`
+ - When opcode is `MUL` (`0x02`) and `res / a != b`
+ - When opcode is `DIV` (`0x04`) or `MOD` (`0x06`); and `b == 0`
+ - When opcode is `ADDMOD` (`0x08`) and `c == 0 ∨ ((a + b) / (2 ** 256) > c)`
+ - When opcode is `MULMOD` (`0x08`) and `c == 0 ∨ ((a * b) / (2 ** 256) > c)`
+ - When opcode is `EXP` (`0x0A`) and ideal `a ** b > 2 ** 256`
+ - When opcode is `SHL` (`0x1b`) and `res >> a != b`
+
+The `serr` flag is "MUST" set in the following circumstances:
+
+ - When opcode is `ADD` (`0x01`) and `a != 0 ^ sgn(a) == sgn(b) ^ sgn(a) != sgn(res)` 
+ - When opcode is `SUB` (`0x03`) and `a != 0 ^ sgn(a) != sgn(b) ^ sgn(b) == sgn(res)`
+ - When opcode is `MUL` (`0x02`) and `(a == -1 ^ b == -(2**255)) ∨ (a == -(2**255) ^ b == -1) ∨ (a != 0 ^ abs(res) / abs(a) != abs(b))`
+ - When opcode is `SDIV` (`0x05`)  or `SMOD` (`0x06`); and `b == 0 ∨ (a == -(2**255) ^ b == -1)`
+ - When opcode is `SHL` (`0x1b`) and `res >> a != b` (this `>>` represents `SAR`)
+
+The function `sgn(num)` returns the sign of the number, it can be negative, zero or positive.
+
+A new opcode, `ARITHERR` (arithmetic error) is added, with number `0x0c`. This opcode takes 0 arguments from the stack. When executed, it pushes `2 * serr + err` and sets both flags to zero.
+
+### gas
+
+The gas cost for the instruction is `G_base` as it is an easily accessible information
+
+## Rationale
+
+EVM uses two's complement for negative numbers. The opcodes listed above triggers one or two flags depending if they are used for signed and unsigned numbers.
+
+The test for each opcode is made with implementation friendlyness in mind. The only exception is EXP as it is hard to give a concise test as most of the others relyed on the inverse operation and there is no native `LOG`. Most `EXP` implementations will internaly use `MUL` so the flag can be drawn from that instruction.
+
+The flag being carryed to the stack is to make it possible to give control to the code implementation on how it will be utilized. It may perform an exceptional halting or any other behavior of it's choosing.
```

Comment:
```suggestion
The flag being carryed to the stack is to make it possible to give control to the code implementation on how it will be utilized. It may perform an exceptional halting or any other behavior of its choosing.
```